mozetl/hardware_report/summarize_json.py

```diff
@@ -19,9 +20,13 @@
 # These fields have a fixed set of values and we need to report all of them.
 EXCLUSION_LIST = ["has_flash", "browser_arch", "os_arch"]
 
-# S3_PUBLIC_BUCKET = "telemetry-public-analysis-2"
 S3_DATA_PATH = "game-hardware-survey/data/"
 
+# Stores all hardware reports in json by date
+_date_to_json = {}
```

Comment:
Please remove this global variable and instead have `generate_report` return the dictionary, which the main function can use in `serialize_results`, while the tests can use it directly.